sencha-workspace/packages/slate-cbl/src/view/student/DashboardController.js

```diff
@@ -0,0 +1,137 @@
+        studentDashboardCompetenciesList.empty();
+        studentDashboardCompetenciesList.removeCls('competencies-unloaded').addCls('competencies-loading');
+
+        contentArea.getCompetenciesForStudents([student.getId()], function(competencies) {
+            var competenciesLength = competencies.length,
+                competencyIndex = 0,
+                competency, skillsList;
+            
+
+            if(!competenciesStore.isLoaded()) {
+                competenciesStore.loadRawData(competencies);
+            }
+
+            competenciesTpl.overwrite(studentDashboardCompetenciesList, {
+                student: student.getData(),
+                competencies: competencies
+            });
+
+            studentDashboardCompetenciesList.removeCls('competencies-loading').addCls('competencies-loaded');
+
+            for (; competencyIndex < competenciesLength; competencyIndex++) {
+                competency = Ext.create('Slate.cbl.model.Competency', competencies[competencyIndex]);
+                skillsList = studentDashboardCompetenciesList.down('.cbl-competency-panel[data-competency="'+competency.getId()+'"] .cbl-skill-meter');     
+                me.loadSkills(student, competency, skillsList);
+            }
+        });
+    },
+    
+    onDemoCellClick: function(dashboardView, ev, targetEl) {
+
+        Ext.create('Slate.cbl.view.student.skill.OverviewWindow', {
+            autoShow: true,
+            animateTarget: targetEl,
+
+            student: window.SiteEnvironment.cblStudent.ID,
+            competency: targetEl.up('ul.cbl-skill-demos').up('li.cbl-competency-panel').getAttribute('data-competency'),
+            skill: targetEl.getAttribute('data-skill')
+        });
+    },
+    
+
+
+    // protected methods
+    loadSkills: function(student, competency, skillsList) {
+        var me = this,
+            skillsTpl = Ext.XTemplate.getTpl(me.view, 'skillsTpl'),
+            skills, demonstrations, _renderSkills;
+
+        skillsList.removeCls('skills-unloaded').addCls('skills-loading');
+
+        _renderSkills = function() {
+            var demonstrationsLength = demonstrations.length, demonstrationIndex = 0, demonstration, skill;
+            // group skills by student
+            for (; demonstrationIndex < demonstrationsLength; demonstrationIndex++) {
+                demonstration = demonstrations[demonstrationIndex];
+                skill = skills.get(demonstration.SkillID);
+
+                if (!skill.demonstrations) {
+                    skill.demonstrations = [];
+                }
+
+                skill.demonstrations.push(demonstration);
+            }
+
+
+            skillsTpl.overwrite(skillsList, skills.items);
+            skillsList.removeCls('skills-loading').addCls('skills-loaded');
+        };
+
+        competency.getDemonstrationsForStudents([student.getId()], function(loadedDemonstrations) {
+            demonstrations = loadedDemonstrations;
+
+            if (skills) {
```

Comment:
`competency.getDemonstrationsForStudents` and `competency.withSkills` are async dataloads fired off in parallel, their callbacks have complementary checks so that _renderSkills() is called once after the second of the two is complete